lib/domain.js

```diff
@@ -47,14 +47,17 @@ const { WeakReference } = internalBinding('util');
 
 // Overwrite process.domain with a getter/setter that will allow for more
 // effective optimizations
-const _domain = [null];
+// Never directly access _domain, use the getter to ensure a consistent hook
+// for user code
+let _domain = null;
```

Comment:
Can you explain why we go from domain arrays to single object here? It feels like this change is unrelated. E.g. I can't see where `configurable` is leveraged.

Reply:
bug from a rebase on another branch ive been dealing with